libs/utils/include/celix_utils.h

```diff
@@ -46,11 +46,42 @@ char* celix_utils_strdup(const char *str);
  */
 unsigned int celix_utils_stringHash(const char* string);
 
+/**
+ * The proposed buffer size to use for celix_utils_writeOrCreateString with a buffer on the stcck.
+ */
+#define CELIX_DEFAULT_STRING_CREATE_BUFFER_SIZE 512
+
+/**`
+ * @brief Format a string to the provided buffer or a newly allocated buffer if the provided buffer is to small.
+ * @param[in,out] buffer The buffer to write the formatted string to.
+ * @param[in] bufferSize The size of the buffer.
+ * @param[in] format The format string.
+ * @param[in] ... The arguments for the format string.
+ * @return The formatted string in the provided buffer or a newly allocated buffer if the provided buffer is to small.
+ */
+char* celix_utils_writeOrCreateString(char* buffer, size_t bufferSize, const char* format, ...)
+    __attribute__((format(printf, 3, 4)));
+
+/**
+ * @brief Free the provided str if the str is not equal to the provided buffer.
+ * @note This function is useful combined with celix_utils_writeOrCreateString.
+ * @param buffer The buffer to compare the str to.
+ * @param str The string to free if it is not equal to the buffer.
+ */
+void celix_utils_freeStringIfNeeded(const char* buffer, char* str);
```

Comment:
celix_utils_freeStringIfNeeded propose to change to celix_utils_freeStringIfNotEqual